### PR TITLE
Update docs links to reference new domain

### PR DESCRIPTION
**Pull request by RealOrangeOne**
As a part of https://github.com/wagtail/wagtail/issues/7841, we need to update which domain is primarily used to receive emails for the Wagtail project.

## To do

- Setup ReadTheDocs with an additional domain
- Deploy updated worker https://github.com/wagtail/wagtail.io/pull/124

### CODE REVIEW

File: scripts/latest.txt [hunk 1]
```diff
@@ -1,10 +1,10 @@
 {
     "version": "2.15",
-    "url":     "https://docs.wagtail.io/en/stable/releases/2.15.html",
```
Comment (RealOrangeOne) [inline scripts/latest.txt:3]: Question: This file looks rather auto-generated, but I couldn't find any other references to where it comes from in the codebase. Do we just mutate the one which already exists?

Comment (gasman) [reply scripts/latest.txt:3]: This is maintained manually and pushed to S3 as part of the release process: https://github.com/wagtail/wagtail/wiki/Creating-a-new-Wagtail-release#releaseswagtailio

(It used to be auto-generated with a command line tool as part of that process, but when we added the LTS / majorUrl bits it made more sense to edit it directly instead - https://github.com/wagtail/wagtail/commit/ec0daf604a366b18c842a13e1821391a5e2abdcd)

Comment (RealOrangeOne) [reply scripts/latest.txt:3]: Ah nice, thanks! Looking at the wiki:

> Edit scripts/latest.txt with the new version details

Looks like the existing process will catch the change so long as these changes are made, which is great!

**Comment by lb-** [comment]
@RealOrangeOne can you rebase this and re-push. Looks like it included formatting changes that were in the other PR

**Review by lb- — APPROVED** [review]
@RealOrangeOne thanks again for going through all these, I have rebased (fixed conflicts) and pushed to you repo.

**Pull request merged**